include/libtorrent/packet_pool.hpp

```diff
@@ -79,24 +79,24 @@ namespace libtorrent
 	};
 
 
-	struct packet_mem_manager
+	struct packet_deleter
 	{
 		// deleter for std::unique_ptr
 		void operator()(packet* p) const
 		{
 			p->~packet();
 			std::free(p);
 		}
-
-		static packet *create_packet(int size)
-		{
-			packet *p = static_cast<packet*>(std::malloc(sizeof(packet) + size));
-			new (p) packet();
-			return p;
-		}
 	};
 
-	using packet_ptr = std::unique_ptr<packet, packet_mem_manager>;
+	static packet *create_packet(int size)
```

Comment:
sorry for nit-picking. I don't think you should define a static function in a header file. Either put it in a .cpp file or make it ``inline``.